rest/docs/design/technical_design.md

```diff
@@ -0,0 +1,127 @@
+At last, you detail how the data will flow across `Connections`, and with which eventual tranformation, specifying a list of `Steps`.
+
+Notice that despite you define an ordered list of `Steps` to define the business behavior of your `Integration` there is no type checking of any sort, between those steps. It's up to you be sure that the output of a `Step` is coherent with the output of the following one.
+
+
+## Integration creation workflow
+
+The entities described and linked above are indeed what the code implementation will use to implement the required behavior. But the idea is that those elements are protected behind some layer of abstraction that allows you not to manipulate them directly.
+
+From a UX point of view, the user is exposed to a graphical UI, dynamically built base on the above models.
+Interacting with a set of forms and UI elements, a user completes the definition of a logical integration.
+
+When she's done, the UI application produces a `.json` files that corresponds to the user activity and that will be used as the input to generate code on his behalf.
+
+The component responsible of consuming this `.json` is the [`ActivateHandler`](https://github.com/syndesisio/syndesis/blob/master/rest/controllers/src/main/java/io/syndesis/controllers/integration/online/ActivateHandler.java), that at runtime, lives in the `syndesis-rest` pod.  
+
+This component is responsible of 3 main activities:  
+1. produce **deplyoment data** 
+1. **build** the project related to our `Integration`
+1. **deploy** an instance of our `Integration`
+
+
+
+##### Build
+**Building** is actually delegated to an implementation of [`ProjectGenerator`](https://github.com/syndesisio/syndesis/blob/master/rest/project-generator/src/main/java/io/syndesis/project/converter/DefaultProjectGenerator.java) that is responsible of **build time code generation**.  
+With some introspection and templating, a full Java based Apache Maven project is generated. The idea is that this artifact could be built succesfully even outside Syndesis runtime platform, being a fully consistent project.  
+The `ActivateHandler`, uses the physical bytes generated by the `ProjectGenerator` to trigger a **build** invocation at **Openshift** level.  
+This means that we leverage **Openshift** "source to image (s2i)" capability, that accepts a reference to a source code project location, to build the project at source code level and produce also a corresponding **Docker** image including everything needed to spawn a runtime instance of our `Integration`.
+
+##### Deploy
+**Deploying**, as another responsibility of `ActivateHandler`. After a source code project corresponding to an `Integration` has been generated, built and shared to the interal repository, it still needs to be deployed to run and be available.
```

Comment:
I think it will b best to write it like that:
**Deploying**, as another responsibility of `ActivateHandler`
--> **Deploying** is another responsibility of `ActivateHandler`